salt/states/pkg.py

```diff
@@ -165,6 +165,10 @@ def _fulfills_version_spec(versions, oper, desired_version,
     otherwise returns False
     '''
     cmp_func = __salt__.get('pkg.version_cmp')
+    # stripping "with_origin" dict wrapper
+    if salt.utils.is_freebsd():
+        if type(versions) is dict and 'version' in versions:
```

Comment:
I amended the commit to switch to isinstance last night.  You can see in "Files changed" tab, but not here in discussion.